### PR TITLE
getMethod option

### PR DESCRIPTION
**Pull request by tameraydin**
Introduces ``getMethod`` option to Twitter provider in order to allow using different methods from its rest API.

### CODE REVIEW

File: lib/providers/twitter.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ exports = module.exports = function (options) {
 
     options = options || {};
```
Comment (ldesplat) [inline lib/providers/twitter.js]: Let's clean this up a bit now that it's getting slightly more complex
```
const defaults = {
    extendedProfile: true,
    getMethod: 'users/show'
};
const settings = Hoek.applyToDefaults(defaults, options);
```

Don't forget to require hoek and change options to settings.

File: lib/providers/twitter.js [hunk 2]
```diff
@@ -20,7 +22,7 @@ exports = module.exports = function (options) {
                 return callback();
             }
 
-            get('https://api.twitter.com/1.1/users/show.json', { user_id: params.user_id }, (profile) => {
+            get('https://api.twitter.com/1.1/' + getMethod + '.json', { user_id: params.user_id }, (profile) => {
```
Comment (ldesplat) [inline lib/providers/twitter.js]: ES6 String
```
get(`https://api.twitter.com/1.1/${settings.getMethod}.json`, ...
```

**Comment by ldesplat** [comment]
Looks good to me, Just these two minor comments. Thanks, now it makes more sense what you were asking for.

**Comment by gergoerdosi** [comment]
I think the name `getMethod` is a bit confusing. Method has a meaning in requests and this setting has nothing to do with that.

**Comment by tameraydin** [comment]
@ldesplat thanks for the feedback.. I did the changes you suggested but now somehow I'm getting this weird "'equal' of undefined" error on CI (when I try locally -node v4,5 or 6-, all tests are passing, no fail). Do you have any idea?

**Comment by gergoerdosi** [comment]
@tameraydin: Remove `deep` in tests, it's the default in code v3.0.0.

**Comment by tameraydin** [comment]
@gergoerdosi thanks, it helped! ..but I'm curious, why the assertion I added was the only one failing, I see that deep comparison is used in all other provider specs & assertions?

**Comment by gergoerdosi** [comment]
I don't see `deep` being used anywhere in the code. Which files are you referring to?

**Comment by tameraydin** [comment]
ahh my bad.. my fork branch wasn't up-to-date:) thx!

**Comment by ldesplat** [comment]
Thank you @gergoerdosi for the consistent review. And thank you @tameraydin for contributing!

**Pull request merged**